### PR TITLE
[IOTDB-31] add props of a time series declarison into FileNodeProcessor

### PR DESCRIPTION
**Pull request by jixuan1989**
* add the `props` of a time series declension into `FileNodeProcessor`

* pick Compressor out of `props`

### CODE REVIEW

**Comment by jixuan1989** [comment]
I change the file format of MManager. So IT IS NOT compatibility with old version. 
YOU HAVE TO REMOVE your mmanager file and register all timeseries again.

File: iotdb/src/main/java/org/apache/iotdb/db/qp/physical/sys/MetadataPlan.java [hunk 1]
```diff
@@ -101,8 +115,8 @@ public void setEncodingArgs(String[] encodingArgs) {
   public String toString() {
     String ret = "seriesPath: " + path + "\ndataType: " + dataType + "\nencoding: " + encoding
         + "\nnamespace type: " + namespaceType + "\nargs: ";
-    for (String arg : encodingArgs) {
-      ret = ret + arg + ",";
+    for (Map.Entry prop : props.entrySet()) {
+      ret = ret + prop.getKey() + "=" +prop.getValue() + ",";
```
Comment (jt2594838) [inline iotdb/src/main/java/org/apache/iotdb/db/qp/physical/sys/MetadataPlan.java]: It is recommended to use StringBuilder where string concatenations are many.

File: iotdb/src/main/java/org/apache/iotdb/db/qp/strategy/LogicalGenerator.java [hunk 2]
```diff
@@ -370,18 +373,36 @@ private void analyzeMetadataCreate(AstNode astNode) throws MetadataArgsErrorExce
...
+      offset++;
+    } else {
+      compressor = TSFileConfig.compressor;
+    }
     checkMetadataArgs(dataType, encodingType);
-    String[] paramStrings = new String[paramNode.getChildCount() - 2];
+    String[] paramStrings = new String[paramNode.getChildCount() - offset];
+    Map<String, String> props = new HashMap<>(paramStrings.length + 1, 1);
+    String[] kv;
+    for (String param : paramStrings) {
```
Comment (jt2594838) [inline iotdb/src/main/java/org/apache/iotdb/db/qp/strategy/LogicalGenerator.java]: When is the content of `paramStrings` modified. I could only see the initialization.

Comment (jixuan1989) [reply iotdb/src/main/java/org/apache/iotdb/db/qp/strategy/LogicalGenerator.java]: Have fixed...

**Comment by MyXOF** [comment]
I have added an IT to test float precision in iotdb/src/test/java IoTDBFloatPrecisionIT.java.
Given a float value 1.2345 and precision = 3, before flushing into disk, it is stored as 1.2345 in memory. After flushing to disk, it is stored as 1.235(after rounding) 

Node that 

```java
      boolean hasResultSet = statement.execute("select * from root");
      ResultSet resultSet = statement.getResultSet();
      int cnt = 0;
      while (resultSet.next()) {
        for(int i = 0; i < 10; i++){
          // original value
          Assert.assertEquals(Float.parseFloat(VALUE), resultSet.getFloat(String.format("root.vehicle.%s.%s", "f0", "s"+i+"rle")), DELTA_FLOAT);
         ........
        }
      }
      // flush to disk
      statement.execute("flush");
      resultSet = statement.executeQuery("select * from root");
      cnt = 0;
      while (resultSet.next()) {
        for(int i = 0; i < 10; i++){
          // value after rounding
          BigDecimal b = new BigDecimal(VALUE);
          Assert.assertEquals(b.setScale(i, RoundingMode.HALF_UP).floatValue(), resultSet.getFloat(String.format("root.vehicle.%s.%s", "f0", "s"+i+"rle")), DELTA_FLOAT);
        }
      }
```

**Pull request merged**